src/stores/useLabyrinthStore.jsx

```diff
@@ -0,0 +1,67 @@
+import { create } from "zustand"
+
+export const useLabyrinthStore = create((set, get) => ({
+  userName: null,
+  loading: false,
+  coordinates: null,
+  description: null,
+  actions: null,
+  error: null,
+  loggedIn: false,
+  startGame: async (name) => {
+    const encryptedName = name + new Date()
+    set({ loading: true, error: null })
+    try {
+      const response = await fetch("https://labyrinth.technigo.io/start", {
+        method: "POST",
+        body: JSON.stringify({ username: encryptedName }),
+        headers: { "Content-Type": "application/json" },
+      })
+      if (!response.ok) {
+        throw new Error("Fetching error")
+      }
+      const data = await response.json()
+      set({
+        userName: encryptedName,
+        coordinates: data.coordinates,
+        description: data.description,
+        actions: data.actions,
+        loggedIn: true,
+      })
+    } catch (error) {
+      set({ error: error })
+    } finally {
+      set({ loading: false })
+    }
+  },
+  makeMove: async (direction) => {
+    set({ loading: true, error: null })
+    try {
+      const response = await fetch("https://labyrinth.technigo.io/action", {
+        method: "POST",
+        body: JSON.stringify({
+          username: get().userName,
+          type: "move",
+          direction: direction,
+        }),
+        headers: { "Content-Type": "application/json" },
+      })
+      if (!response.ok) {
+        throw new Error("Movement fails...")
+      }
+      const data = await response.json()
+      set({
+        coordinates: data.coordinates,
+        description: data.description,
+        actions: data.actions,
```

Comment:
Would be nice if you also saved the description of the actions, that's quite a big part of the game 👀